static/colors.css

```diff
@@ -0,0 +1,96 @@
+/* --------------------------------------------------------------------------
+ * DARK MODE ---------------------------------------------------------------- */
+
+:root {
+    --bg: black;
+    --bg-highlight: #161616;
+    --bg-code: #333;
+    --bg-mark: rgba(255, 255, 175, 0.8); 
+
+    --text-primary: #aaa;
+    --text-secondary: #6a6a6a; /* #7d7d7d from foot notes */
+    --text-link: #859abc;
+    --text-mark: #222;
+
+    --border-light: #444;
+    --border-dark: #353535;
+    --border-kbd: #242424;
+
+    --syntax-text: var(--text-primary);
+    --syntax-keyword: #aed7ff;
+    --syntax-entity: #aeafff;
+    --syntax-constant: #d1b0fa;
+    --syntax-string: #d787af;
+    --syntax-symbol: var(--syntax-constant);
+    --syntax-comment: #949494;
+    --syntax-entity-tag: #d7afd7;
+    --syntax-storage-modifier-import: var(--syntax-text);
+
+    --syntax-markup-heading: var(--syntax-text);
+    --syntax-markup-list: var(--syntax-text);
+    --syntax-markup-italic: var(--syntax-text);
+    --syntax-markup-bold: var(--syntax-text);
+    --syntax-markup-inserted-fg: #aff5b4;
+    --syntax-markup-inserted-bg: #033a16;
+    --syntax-markup-deleted-fg: #ffdcd7;
+    --syntax-markup-deleted-bg: #67060c;
+
+    --alert-note: #a5d5fe;
+    --alert-tip: #b4fa72;
+    --alert-important: #ff8ffd;
+    --alert-warning: #ffaf00;
+    --alert-caution: #ff5f5f;
+
+    --ipynb-bg-error: rgba(255, 0, 0, 0.1);
+}
+
+/* --------------------------------------------------------------------------
+ * LIGHT MODE --------------------------------------------------------------- */
+
+@media (prefers-color-scheme: light) {
+    :root {
+        --bg: white;
+        --bg-highlight: #f6f8fa;
+        --bg-code: #afb8c133;
+        --bg-mark: rgba(255, 255, 175, 1);
+
+        --text-primary: #1f2328;
+        --text-secondary: #656d76;
+        --text-link: #0969da;
+        --text-mark: var(--text-primary);
+
+        --border-light: #d8dee4;
+        --border-dark: #d0d7de;
```

Comment:
Naming works for dark theme, but for light theme these are essentially vice-versa: light is dark and dark is light

Some alternative suggestions:

--border / --border-dimmed
--border / --border-faint
--border-regular / ...

Reply:
Haha funny, I actually did this by just giving the **light** mode media query part of the original `styles.css` to GPT4 and having it find names for the colors. I modified some names so they are more structured, but `border-light` was one I left the way GPT named it. Then later when I did the dark mode colors I checked if `border-light` would "also" be the lighter one in dark mode, surprisingly saw that it was, and assumed everything would fit :D

Reply:
How about we take the naming from how font weights are named?

https://developer.mozilla.org/en-US/docs/Web/CSS/font-weight#common_weight_name_mapping

I would suggest

```txt
border-light → border-light
border-dark  → border-medium
border-kbd   → border-bold
```

We could also name `border-light` as `border-regular` because that is the most "regular" border color, but I'm not sure how clear it is to people that "medium" is stronger than "regular". Or another option might be sticking more to the applicable [CSS values](https://developer.mozilla.org/en-US/docs/Web/CSS/font-weight#values):

```txt
border-light → border-normal
border-dark  → border-bold
border-kbd   → border-bolder
```

OR taking something from [Tailwind's naming](https://tailwindcss.com/docs/font-weight):

```txt
border-light → border-normal
border-dark  → border-semibold
border-kbd   → border-bold
```

Maybe this is the most sensible solution? I think if anyone knows that they are doing with naming style-related things it's Tailwind ^^

Reply:
Hmmmm it's complicated

One thing I notice that on light theme the kbd border certainly doesn't correlate with `bold`, rather it's the faintest border I've ever seen in my life:

![image](https://github.com/user-attachments/assets/5369d540-3c7e-4471-b513-36d5d91cf3dd)

I think it's tricky because I'd say all border colors we have are very faint but slightly different.

Other than that, good ideas. Just I'm not sure which one should be considered the `normal` one, is it the <h1> bottom border color?

Reply:
We could also consider putting the whole border definition into the variable. E.g. make the value `0.5px solid #...`. This way the naming by "bold" and so on would be easier and the border width becomes part of the themes

Reply:
Oh, is the width what you were thinking about with `bold` I had the darkness/lightness of color in my mind still :smile:

Reply:
No haha, I was thinking about both. So maybe it's good to include both. Also because a different theme with a different color might need a different width to look good

Reply:
Yeah, I worry about a slippery slope towards a mess though if we include variables other than colors (at least right now)?

![image](https://github.com/user-attachments/assets/ca747565-1eba-4e8d-88ed-671d890ec066)

Here are the border colors side-by-side btw,

My perception is like (left-to-right):

* normal
* slightly more pronounced
* a special case for kbd

Reply:
Not really any solution, just more overview:

Colors for both themes (same order, currently "light" → "dark" → "kbd"):

<img width="1512" alt="image" src="https://github.com/user-attachments/assets/65ec4bfb-a62e-4cf1-a53a-9873f643dd2b">

How it looks in practice

<img width="266" alt="image" src="https://github.com/user-attachments/assets/72408e1a-c5ab-40f5-a089-a7145a771ec5">
<img width="338" alt="image" src="https://github.com/user-attachments/assets/58c68ae8-b44d-4a94-a3c8-7f56b0332c13">

Markdown to reproduce:

```markdown
## header

<kbd>kbd</kbd>

> block <kbd>kbd</kbd>

---

| table[^1] |
| :-- |
| <kbd>kbd</kbd> |
| <kbd>kbd</kbd> |
| |
| |

[^1]: footnote
```

Reply:
As a follow-up on my idea to include the whole border style, here are the distince styles we currently have[^1]:

```css
h1, h2, section.footnotes, dd /*definition lists*/ {
  border: 0.2px solid var(--border-light);
}

blockquote {
  border: 0.25em solid var(--border-dark);
}

hr /*horizontal rule*/ {
  border: 24px solid var(--border-dark);
}

th, td /*tables*/ {
  border: 1px solid var(--border-light);
}

kbd {
  border: 1px solid var(--border-kbd);
}
```

Looking at this I think we could unify `blockquote` and `hr` to use the same style, and also the tables and `kbd`. This would give us more consistent styling and maybe make it easy to name all 3 styles.

[^1]: note that I wrote `border` everywhere even though it might be `border-left` or another side for some examples but this wouldn't break from using the variables as I suggested

Reply:
Ok, it's tripping me up big time that I chose a color back in #58 opposite to how I'd expect it to be as a light-theme counterpart, but it still looks appropriate to me and maybe it was deliberate.

\<kbd> doesn't look good in alternate table cell but not too unexpected and I suppose GitHub has the same issue?

Reply:
Okay so including the whole border definition in 1 variable doesn't work well because we need *only* the color to make the shadow for the `kbd`-element. So back to just colors. I still liked the idea of getting rid of one of the 3 colors though and made a suggestion now in the latest commit and as described below.

**Suggestion** We keep only 2 border colors: `border-regular` for slim borders less than 1px and `border-bold` for borders from 1px. The screenshots below show the before/after of this change, left is before, right is after.

<img width="437" alt="image" src="https://github.com/user-attachments/assets/4fd49e14-43e5-4ff2-8ec2-a91db0c37071">
<img width="434" alt="image" src="https://github.com/user-attachments/assets/e25f9303-b6af-4354-887c-e521c28a91f1">

I particularly like (1) that the `kbd` element has much better visibility for light mode now, and (2) the table borders aren't as overly contrasty anymore for dark mode. Additionally, we would of course reduce complexity of the CSS/colors this way since we get rid of the specialized color that is only used for the `kbd` element.

What do you think?

Reply:
I'm kinda wary about changing some of the border colors from what they were on GitHub when they look just fine as is, but also have a bit of a dilemma because the border you suggest for \<kbd> looks much better, no doubt...

Maybe I can word what I'm thinking like this:

* We don't have to follow GitHub's style to a tee, when their style is not good (their \<kbd> border is so faint you can hardly see it)
* When there's nothing wrong with the GitHub style (e.g. light theme table border *imo*), not sure we should change that :thinking: 

I still feel like there should be 2 border colors that can be applied freely to anything without any regard to border thicknesses, for example on GH these same thickness borders are like (with their var names):

![image](https://github.com/user-attachments/assets/ae3049f9-6a60-4665-a821-9ef846b22eba)

However it's further complicated by that I agree with you that the fainter table border looks better on dark theme. How would that fit into this then...

I have to really properly investigate this but if this gives you any thoughts and if we have a disagreement, let me know :smile:

Reply:
But one thing I'm sure: removal of border-kbd variable in favor of something that looks better on it is an absolute +1 from me

Reply:
Hm, I don't really have the goal of following GitHub's style at all to be honest haha, our dark theme is also completely different anyways. What's important for me is to have something that (1) looks good with everything being well-visible, and (2) is as simple as possible because (other than GitHub) we are not a huge team working full time on this and are (at least talking for myself) not professional designers :D

Another argument for favoring simplicity is that the state of this branch makes it very easy to create a Vivify color theme (i.e. a CSS file that just defines color variables) and creating one for GitHub is insanely hard and complicated because of the huge mountain of hard-to-understand colors that they define. I get taking GitHub as a kind of "gold standard" but I think its file-viewing goals diverge from those of Vivify in some important aspects (such as customizability).

And, finally, if our goal is to follow GitHub's styles, that will also mean we have to regularly check *everything* we take from their styles to make sure we are still up-to-date.

---

So, long story short (sorry if that was too much of a rant ^^) I would merge this as it is, but if you have another idea that takes my arguments into account, please feel free to suggest something different!

Reply:
> I still feel like there should be 2 border colors that can be applied freely to anything without any regard to border thicknesses, for example on GH these same thickness borders are like (with their var names):

We can also still do this, just have to come up with names. GitHub's "-muted" might be an option. Like I said, please make suggestions if you have ideas :) I'm just starting to run out of ideas for this haha and would like to merge soon to do #101 based on this

Reply:
Yeah no worries, I think we can merge today

> And, finally, if our goal is to follow GitHub's styles, that will also mean we have to regularly check *everything* we take from their styles to make sure we are still up-to-date.

This is a big point that was inevitably going to catch up to us, in fact if I'm not wrong some of the colors may have already been slightly adjusted... and I was wondering whether we aim to follow or what, and yeah definitely lets not try to keep up with that.

I was confused above with borders `<1px` because they end up being `1px` on my machine so I thought the changes here meant that h1 bottom border and table border would be the same, but it's not the case. So actually light mode is just fine!

How do you feel about this discrepancy:

* On light mode, regular is muted (less contrast)
* On dark mode however, bold is the muted one

^ a decision still resulting from my choice of color in #43 for quote border

So if I fix this logical inconsistency, for example in this way:

* border-regular: `#444444` --> `#303030`
* border-bold: `#353535`

^ (keeping the border color we currently like and readjusting regular around that)

This is how it would look like:

![image](https://github.com/user-attachments/assets/711f6b29-79db-4a55-929a-781d66a40baf)

^ With that, we'd have the same definition for border-regular and border-bold regardless of light/dark

Do you think we wanna make this change?

Reply:
Yes, good point, I was actually going to suggest the same but adjusting the light mode colors the other way around instead😂 Made more sense to me that thicker borders can be muted and you still see them well whereas thing borders need more contrast to be visible. But of course the other way around it also makes sense, i.e. thinner borders are supposed to be more subtle so they have a more muted color.

If you want, check out how light mode looks if you make the same change the other way around, pick the option you prefer, commit it, approve the PR and then we merge☺️

Reply:
Ah, lets see.

So just to be clear, you think the one I suggested above looks good to you on dark?

Reply:
By the way, sorry this is super off-topic here, but could you maybe also check #158? haha. The newest release has an issue where vivify-server has an undefined port when there is no custom config so it crashes on startup. For me #158 fixes it but I have yet to get a confirmation from the people on the corresponding issue🙈 So in case you are quicker and don't mind maybe you could check that ^^

Reply:
> So just to be clear, you think the one I suggested above looks good to you on dark?

Yes, I'm fine with either option :) Feel free to decide